internal/auth/user.go

```diff
@@ -9,19 +9,19 @@ import (
 type usernameKeyType struct{}
 
 var (
-	usernameKey usernameKeyType
+	UsernameKey usernameKeyType
```

Comment:
Why do you change this to public?
You can use "UserFromContext" function instead, no?
Anyway, I don't think that changing encapsulation level of variable only for tests usage is good solution

Reply:
yeah. I think it's better to expose `newContext` instead of the key.
> You can use "UserFromContext" function instead, no?
It's about injecting the struct in context.